cmd/neofs-cli/internal/client/client.go

```diff
@@ -303,29 +306,93 @@ func (x *PutObjectPrm) SetPayloadReader(rdr io.Reader) {
 
 // PutObjectRes groups resulting values of PutObject operation.
 type PutObjectRes struct {
-	cliRes *client.ObjectPutRes
+	id *oidSDK.ID
 }
 
 // ID returns identifier of the created object.
 func (x PutObjectRes) ID() *oidSDK.ID {
-	return x.cliRes.ID()
+	return x.id
 }
 
 // PutObject saves the object in NeoFS network.
 //
 // Returns any error prevented the operation from completing correctly in error return.
-func PutObject(prm PutObjectPrm) (res PutObjectRes, err error) {
-	var putPrm client.PutObjectParams
+func PutObject(prm PutObjectPrm) (*PutObjectRes, error) {
+	var putPrm client.PrmObjectPutInit
 
-	putPrm.WithObject(prm.hdr)
-	putPrm.WithPayloadReader(prm.rdr)
+	wrt, err := prm.cli.ObjectPutInit(context.Background(), putPrm)
+	if err != nil {
+		return nil, fmt.Errorf("init object writing: %w", err)
+	}
 
-	res.cliRes, err = prm.cli.PutObject(context.Background(), &putPrm, append(prm.opts,
-		client.WithSession(prm.sessionToken),
-		client.WithBearer(prm.bearerToken),
-	)...)
+	if prm.sessionToken != nil {
+		wrt.WithinSession(*prm.sessionToken)
+	}
 
-	return
+	if prm.bearerToken != nil {
+		wrt.WithBearerToken(*prm.bearerToken)
+	}
+
+	if prm.local {
+		wrt.MarkLocal()
+	}
+
+	if wrt.WriteHeader(*prm.hdr) {
+		sz := prm.hdr.PayloadSize()
+
+		if data := prm.hdr.Payload(); len(data) > 0 {
+			if prm.rdr != nil {
+				prm.rdr = io.MultiReader(bytes.NewReader(data), prm.rdr)
+			} else {
+				prm.rdr = bytes.NewReader(data)
+				sz = uint64(len(data))
+			}
+		}
+
+		if prm.rdr != nil {
+			const defaultBufferSizePut = 4096 // configure?
```

Comment:
Could you also create a task for this and link to in from here (may be referencing other places with such restrictions)?